### DISCUSSION TITLE
Extending Chakra Types in order to add custom fields?

### DISCUSSION THREAD

**Discussion by hems** [chakra-ui/chakra-ui#2288] [Q&A] [answered]
Hello, i'm trying to create a component that returns a `Flex` Element wrapping an Icon, naturally i would like to keep the Auto Complete from the `Flex Type` when using my component but additionally i would like to add an extra prop to it called "iconName".

This is the code i wrote:

```ts
type TFlex = typeof Flex

const Tab: TFlex = ({ iconName, ...props }) => {
  return (
    <Flex align="center" height="12" minWidth="12" {...props}>
      <Icon name={iconName} size="8" />
    </Flex>
  )
}

export default Tab
```

But doing so is yielding the following error:

![image](https://user-images.githubusercontent.com/27327/96860936-4f4da480-145b-11eb-9aa7-df7f2320b78b.png)

From my understanding I would need to:

 - Extend the `FlexProps Inteface` and call it MyFlexProps
 - Extend the `Flex Type` and call it MyFlexType
 - Type my `Tab` instance with MyFlexType

But unfortunately i could not make this work and also i'm not sure if that's the right approach?

Thank you very much for your help.

**Comment by hems** [comment]
Turns out this solved my issue:

````ts
interface ITab extends FlexProps {
  iconName?: string
}

const Tab: React.FC<ITab> = ({ iconName, ...props }) => {
  return (
    <Flex align="center" height="12" minWidth="12" {...props}>
      <Icon name={iconName} size="8" />
    </Flex>
  )
}
````

**Reply by igdev116** [reply]
Thanks, I fixed my issue ✨🎉🎉

**Reply by Wonder0xWeird** [reply]
This was the magic ✨

**Comment by ljosberinn** [comment, marked as answer]
This is indirectly related to [the as prop and custom components](https://next.chakra-ui.com/guides/as-prop) and I'd kindly suggest using Option 1 as its the least invasive for TS.

**Reply by hems** [reply]
> This is indirectly related to [the as prop and custom components](https://next.chakra-ui.com/guides/as-prop) and I'd kindly suggest using Option 1 as its the least invasive for TS.

awesome, that looks much cleaner, thank you!

**Reply by hems** [reply]
@ljosberinn the link is broken now? maybe there is an updated link?

**Reply by ljosberinn** [reply]
https://chakra-ui.com/guides/as-prop there you go!
